Description: Make sure login do not throw exception for unknown users.
 Do not expose that the tested login had an unknown user.
Author: Petter Reinholdtsen <pere@hungry.com>

Forwarded: https://github.com/NickDaly/Plinth/pull/33
Reviewed-By: Petter Reinholdtsen <pere@hungry.com>
Last-Update: 2013-09-27

--- plinth-0.0.0~git.20130925.orig/modules/installed/lib/auth.py
+++ plinth-0.0.0~git.20130925/modules/installed/lib/auth.py
@@ -27,7 +27,10 @@ def check_credentials(username, passphra
         cfg.log(error)
         return error
 
-    u = cfg.users[username]
+    if username in cfg.users:
+        u = cfg.users[username]
+    else:
+        u = None
     # hash the password whether the user exists, to foil timing
     # side-channel attacks
     pass_hash = hashlib.md5(passphrase).hexdigest()
